src/core/validate.cpp

```diff
@@ -694,87 +1129,44 @@ namespace validate
+        : m_base_url(url)
     {
-        std::string meta_type;
-        try
-        {
-            meta_type = j.at("_type");
-        }
-        catch (...)
-        {
-            LOG_ERROR << "'_type' not found in 'root' metadata";
-            throw role_metadata_error();
-        }
-        if (meta_type != role.type())
-        {
-            LOG_ERROR << "Wrong '_type' found in 'root' metadata, should be 'root': '"
-                      << j.at("_type") << "'";
-            throw role_metadata_error();
-        }
-
-        try
+        if (spec_version == SpecVersion::kV06)
         {
-            from_json_base(j, role);
-            role.set_keys(j.at("keys").get<std::map<std::string, Key>>());
-            role.set_roles(j.at("roles").get<std::map<std::string, RoleKeys>>());
+            p_root = std::make_unique<v1::RootRole>(local_trusted_root);
         }
-        catch (const json::exception& e)
+        else if (spec_version == SpecVersion::kV1)
         {
-            LOG_ERROR << "Error while loading 'root' metadata from JSON file: " << e.what();
-            throw role_metadata_error();
+            p_root = std::make_unique<v1::RootRole>(local_trusted_root);
         }
-        catch (...)
+        else
         {
-            LOG_ERROR << "Error while loading 'root' metadata from JSON file";
-            throw role_metadata_error();
-        }
-
-        // Did not managed to use Role enum as map key
-        // TODO: replace those checks with JSON schema?
-        for (const auto& r : role.roles())
-        {
-            json jr = r.first;
-            auto role = jr.get<Role>();
-            if (role == Role::kInvalid)
-            {
-                LOG_ERROR << "Invalid role while loading 'root' metadata from JSON file: '"
-                          << r.first << "'";
-                throw role_metadata_error();
-            }
+            throw spec_version_error();
         }
+    }
 
-        std::set<std::string> current_roles;
-        for (const auto& r : role.roles())
-        {
-            current_roles.insert(r.first);
-        }
-        std::set<std::string> mandatory_roles = { "root", "snapshot", "targets", "timestamp" };
-        if (!std::includes(current_roles.begin(),
-                           current_roles.end(),
-                           mandatory_roles.begin(),
-                           mandatory_roles.end()))
-        {
-            std::vector<std::string> diff;
-            std::set_difference(mandatory_roles.begin(),
-                                mandatory_roles.end(),
-                                current_roles.begin(),
-                                current_roles.end(),
-                                std::inserter(diff, diff.end()));
-            LOG_ERROR << "Missing roles while loading 'root' metdata: " << mamba::join(", ", diff);
-            throw role_metadata_error();
-        }
+    void RepoTrust::update_all_roles()
+    {
+        // p_root->update();
```

Comment:
?

Reply:
maybe this function should be removed if it's not implemented yet? :)

Reply:
sure it will be done